### PR TITLE
support for 'update_cache' in pkgin module

### PR DESCRIPTION
**Pull request by jasperla**
##### Issue Type:
- Feature Pull Request
##### Plugin Name:

pkgin
##### Ansible Version:

```
ansible 2.0.1.0
  config file =
  configured module search path = Default w/o overrides
```
##### Summary:

`pkgin` uses a locally cached copy of a packages database to lookup package names, and so forth. Currently the `pkgin` module has no way to update this database. Common practise in other modules is through a `update_cache` option, which is now implemented for `pkgin` too.
##### Example output:

```
TASK [update cache] ************************************************************
changed: [host] => {"changed": true, "msg": "updated repository database"}
```

cc'ing @L2G for extra review as he most recently worked on this module

### CODE REVIEW

File: packaging/os/pkgin.py [hunk 1]
```diff
@@ -208,19 +227,34 @@ def install_packages(module, pkgin_path, packages):
...
+        module.fail_json(msg="could not update package db")
 
 
 def main():
     module = AnsibleModule(
             argument_spec    = dict(
                 state        = dict(default="present", choices=["present","absent"]),
-                name         = dict(aliases=["pkg"], required=True)),
+                name         = dict(aliases=["pkg"]),
+                update_cache = dict(default='no', choices=BOOLEANS, type='bool')),
```
Comment (resmo) [inline packaging/os/pkgin.py]: remove `choices=BOOLEANS`, its deprecated. `type=bool` only should be used.

**Comment by resmo** [comment]
Please update the doc for arg `name`: remove `required: true` and add `default: null`

File: packaging/os/pkgin.py (hunk 1 above)
Comment (jasperla) [inline packaging/os/pkgin.py]: ack.

**Comment by jasperla** [comment]
@resmo I've updated this PR with your feedback, thanks.

**Comment by gregdek** [comment]
Thanks @jasperla. @L2G @szinck please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by troy2914** [comment]
@jasperla per the guidelines,  `required:` should have been changed to `false`, vs removing it.

**Comment by jasperla** [comment]
@troy2914 ok, so then I'll re-add `required` back and keep the `default: null` ?

**Comment by troy2914** [comment]
yep, from the guidelines:
"required should always be present, be it true or false
If required is false you need to document default, even if the default is ‘null’ (which is the default if no parameter is supplied). Make sure default parameter in docs matches default parameter in code."

**Comment by jasperla** [comment]
@troy2914 thanks, it's updated now. I've got a whole bunch of other features for this provider ready when this PR has been merged (`upgrade`, `full_upgrade`, `clean` and some refactoring).

**Comment by resmo** [comment]
Thanks @jasperla for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

**Comment by jasperla** [comment]
Ping?

**Comment by resmo** [comment]
@jasperla yes? did you see my comment? It is still not mergeable. Let me know if you need assistance.

**Comment by jasperla** [comment]
Silly me, I completely missed your comment.

**Comment by jasperla** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @jasperla for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by jasperla** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @jasperla. To the current maintainers, @L2G, @szinck please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by jasperla** [comment]
@gregdek what's the policy if neither maintainers respond in a timely manner? I'd love to move forward with this module.

**Comment by gregdek** [comment]
Typically, we give maintainers a couple of weeks to respond before we politely remind them, and a couple of weeks after that we start to look for additional reviewers. 

In this case, I'm happy to hurry it along a bit. @troy2914, I see that you gave some useful reviews; thanks for them. Would you like to consider becoming an official maintainer for this module? It would basically consist of doing the same thing you already did for this module. :)

**Comment by troy2914** [comment]
Is there some documentation on expectations for an official maintainer for this module?  (beyond the module-checklist)  I've read the nice flow chart in REVIEWERS.md too.

**Comment by gregdek** [comment]
Yes: 

http://docs.ansible.com/ansible/developing_modules.html#module-checklist

Each time there's a PR to a module you maintain, you should get a ping from the bot asking for your review and reminding you of how to do it (see comment 3 above). 

What do you say? :)

**Comment by troy2914** [comment]
Okay, I accept.

**Comment by gregdek** [comment]
OK, you're in. :)

Next bot run (6 hours-ish) and it should be official, and your next "shipit" will promote this PR for potential inclusion.

**Comment by gregdek** [comment]
And thank you!

**Comment by troy2914** [comment]
Note I am not sure the `version_added: "2.1"` is correct, given the current version of the development is listed in VERSION as "2.0.0-0.5.beta3"?

**Comment by gregdek** [comment]
Thanks @jasperla. To the current maintainers, @L2G, @szinck, @troy2914 please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by troy2914** [comment]
shipit

**Comment by gregdek** [comment]
Thanks again to @jasperla for this PR, and thanks @L2G, @szinck, @troy2914 for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by jasperla** [comment]
@gregdek @troy2914 thanks guys, I really appreciate your efforts!

**Pull request merged**